doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,345 @@
+  - Control temperature, light, ventilation, etc
+  - Connect to the building management system, via wired or wireless connectors
+  - Deliver data with reasonable timing and accuracy.
+
+## 2.6. Smart home devices
+
+Easy to use devices can let consumers monitor and control their home
+automatically. The devices need to be able to:
+
+  - Sense temperature, light, humidity, smoke, etc
+  - Integrate with home appliances to control temperature, light, power usage,
+    etc
+  - Be usable but secure for people with no technical knowledge 
+  - Connect to a commercial home gateway 
+  - Communicate over common home IoT protocols such as ZigBee, Bluetooth, WiFi,
+    etc
+  - Connect to the building's power or provide their own power, depending on
+    the product
+  
+## 2.7. Daughterboards
+
+Plug-in boards can give devices immediate support for a protocol or standard,
+or let them outsource a task from the main processor. This requires the board
+to:
+
+  - Perform its promised task reliably
+  - Integrate and communicate reliably with the main processor
+  - Include and support any relevant hardware
+
+## 2.8. Education
+
+The broad technical scope of RIOT makes it useful as a basis for education.
+This requires:
+
+  - The presence of didactic materials related to RIOT
+  - Clear and easy to find documentation
+  - A general availability of supported hardware and tools
+
+# 3. Design philosophies
+
+RIOT satisfies the requirements of all the use cases given above, and does so
+in its own unique way, a way that reflects the collective personality of the
+RIOT community. RIOT is the Friendly IoT Operating System: it is friendly to
+users, allowing them to easily build and deploy IoT solutions that are stable
+and trouble-free.
+
+Below are the design philosophies that are typically followed by developers to
+achieve this, including descriptions of tradeoffs between them and where the
+resolutions typically fall.
+
+## Suitability for constrained devices
+
+"Constrained" means that available memory, energy, and processor cycles are so
+reduced as to become a dominant consideration in design requirements [1]. 
+
+#### Energy efficiency
+
+RIOT nodes sometimes need to last for several years without external power, so
+they need to manage energy carefully. RIOT's tickless scheduler lets devices
+sleep while they aren't collecting data.
+
+Modules outside the core should leverage the benefits and address the
+programming challenges of such a scheduler. They shouldn't demand that users do
+the same.  They should, however, allow users to manage power through different
+modes and functions.
+
+#### Small memory footprint
+
+Most of RIOT's targeted use cases are well addressed by devices in class 1 of
+the taxonomy presented in [1]. If small price differences are important or the
+energy budget is particularly tight, the available memory might be near the
+bottom of this class. Over-the-air updating currently reduces the available ROM
+by over half.
```

Comment:
I would propose to remove the last sentence, because this is just a comment on the current state of development.

Reply:
I think this is important to consider in the context of this section, which should serve as a rough guideline for our memory footprint. I've rephrased it as "Over-the-air updating facilities may also reduce the available RIOT by over half, depending on the architecture". Is this more suitable?

Reply:
Don't think so, because you could have implementations that read updates from the network in a tiny bootloader etc. 
Making the case that something needs a certain amount of memory needs to be on firm grounds ... such as: We cannot supply a buffer of 1K in less than 1K memory.

Reply:
This suggestion was originally made by @haukepetersen . Have talked with him offline about this rationale above, we both agreed, removed